### PR TITLE
Add Canvas to RC

### PR DESCRIPTION
**Pull request by profxj**
Provides access to the Canvas of a given Channel within the RC plugin.
Includes a new canvas proxy within ginga.util.grc

Also added docs.  

Can add a Test if given guidance.  
I have confirmed it performs as intended in my own applications.

### CODE REVIEW

File: doc/manual/plugins_global/rc.rst [hunk 1]
```diff
@@ -110,3 +110,72 @@ usually interpreted as a program option.  In order to pass a signed
...
+
+One can load an image from memory in a channel of
+one's choosing.  First connect to a Channel::
+
+    ch = viewer.channel('Image')
+
+Then load a numpy image (i.e. any 2D ndarray)::
+
+    img = np.ones((100,100))
+    ch.load_np('any_str_here', img, 'fits', {})
```
Comment (ejeschke) [inline doc/manual/plugins_global/rc.rst]: On python 3 this gives me the following error:
```python
Fault: <Fault 1: "<class 'TypeError'>:object of type 'Binary' has no len()">
```
This particular one looks like my issue with marshaling numpy arrays correctly under python 3...

File: ginga/rv/plugins/RC.py [hunk 2]
```diff
@@ -346,5 +346,23 @@ def ginga(self, method_name, *args, **kwdargs):
...
+    def canvas(self, chname, command, *args, **kwargs):
+        chinfo = self.fv.get_channel(chname)
+        canvas = chinfo.viewer.get_canvas()
+        if command == 'add':
+            klass = canvas.get_draw_class(args[0])
+            newargs = args[1:]
+            obj = klass(*newargs, **kwargs)
+            return self.fv.gui_call(canvas.add, obj)
+        elif command == 'clear':
+            canvas.delete_objects()
```
Comment (ejeschke) [inline ginga/rv/plugins/RC.py]: I think we need an "else" clause here, probably raising an error if the command is not one of the recognized ones.

File: ginga/rv/plugins/RC.py [hunk 3]
```diff
@@ -346,5 +346,23 @@ def ginga(self, method_name, *args, **kwdargs):
...
+
+    #def add_canvas_obj(self, chname, typename, *args, **kwargs):
+    #    chinfo = self.fv.get_channel(chname)
+    #    canvas = chinfo.viewer.get_canvas()
+    #    klass = canvas.get_draw_class(typename)
+    #    obj = klass(*args, **kwargs)
+    #    return self.fv.gui_call(canvas.add, obj)
+
```
Comment (ejeschke) [inline ginga/rv/plugins/RC.py]: Can this commented code be removed if it is not being used?

File: doc/manual/plugins_global/rc.rst [hunk 4]
```diff
@@ -110,3 +110,72 @@ usually interpreted as a program option.  In order to pass a signed
...
+Or add any basic Canvas object.  The key issue to keep in
+mind is that the objects input must pass through the XMLRC
+protocol.  This means simple data types:  float, int, lists, str.
+No arrays.  Here is an example to plot a line through a series
+of points defined by two Numpy arrays:
+
+    x = np.arange(100)
+    y = np.sqrt(x)
+    points = zip(x.tolist(), y.tolist())
+    canvas.add('path', points, color='red')
```
Comment (profxj) [inline doc/manual/plugins_global/rc.rst]: Ah, good to know.  Added.

File: doc/manual/plugins_global/rc.rst (hunk 1 above)
Comment (profxj) [inline doc/manual/plugins_global/rc.rst]: Should we raise it as an issue?

File: doc/manual/plugins_global/rc.rst [hunk 5]
```diff
@@ -110,3 +110,72 @@ usually interpreted as a program option.  In order to pass a signed
...
+
+It is possible to add objects to the Canvas in a given
+Channel.  First connect::
+
+    canvas = viewer.canvas('Image')
+
+This connects to the Channel named "Image".  One can
+clear the objects drawn in the Canvas::
+
+    canvas.clear()
```
Comment (profxj) [inline doc/manual/plugins_global/rc.rst]: Strange..  I am getting that now too and hadn't been.  Fixing..

Comment (profxj) [inline doc/manual/plugins_global/rc.rst]: Ok, figured it out. My clear now attempts to only remove
the added objects and not the image (which is removed by
delete_all_objects).  If you have something more elegant, please suggest it.

File: ginga/rv/plugins/RC.py (hunk 2 above)
Comment (profxj) [inline ginga/rv/plugins/RC.py]: Added

**Comment by ejeschke** [comment]
@profxj , testing the latest changes with python 3 ...

**Comment by ejeschke** [comment]
Merged.  Also fixed the buffer encoding issue under python 3, I mentioned above.

**Pull request merged**